ee/codegen/src/__test__/nodes/tool-calling-node.test.ts

```diff
@@ -155,51 +154,67 @@ describe("ToolCallingNode", () => {
   });
 
   describe("composio tool", () => {
-    const composioToolFunction: ComposioToolFunctionArgs = {
-      type: "COMPOSIO",
-      name: "github_create_an_issue",
-      integration_name: "GITHUB",
-      tool_slug: "GITHUB_CREATE_AN_ISSUE",
-      description: "Create a new issue in a GitHub repository",
-    };
-
-    it("should generate composio tool", async () => {
-      const nodePortData: NodePort[] = [
-        nodePortFactory({
-          id: "port-id",
-        }),
-      ];
-
-      const functionsAttribute = nodeAttributeFactory(
-        "functions-attr-id",
-        "functions",
-        {
-          type: "CONSTANT_VALUE",
-          value: {
-            type: "JSON",
-            value: [composioToolFunction],
-          },
-        }
-      );
-
-      const nodeData = toolCallingNodeFactory({
-        nodePorts: nodePortData,
-        nodeAttributes: [functionsAttribute],
-      });
-
-      const nodeContext = (await createNodeContext({
-        workflowContext,
-        nodeData,
-      })) as GenericNodeContext;
-
-      const node = new GenericNode({
-        workflowContext,
-        nodeContext,
-      });
-
-      node.getNodeFile().write(writer);
-      expect(await writer.toStringFormatted()).toMatchSnapshot();
-    });
+    it.each([
+      {
+        name: "with integration_name and tool_slug",
+        composioToolFunction: {
+          type: "COMPOSIO",
+          name: "github_create_an_issue",
+          integration_name: "GITHUB",
+          tool_slug: "GITHUB_CREATE_AN_ISSUE",
```

Comment:
[oos] let's add a comment about how this is the legacy shape and not expected to be maintained long term